src/attributes/codegen.md

```diff
@@ -52,15 +52,51 @@ r[attributes.codegen.naked]
 r[attributes.codegen.naked.intro]
 The *`naked` [attribute]* prevents the compiler from emitting a function prologue and epilogue for the attributed function.
 
+> [!EXAMPLE]
+> ```rust
+> # #[cfg(target_arch = "x86_64")] {
+> /// Adds 3 to the given number.
+> ///
+> /// SAFETY: The validity of the used registers
+> /// is guaranteed according to the "sysv64" ABI.
+> #[unsafe(naked)]
+> pub extern "sysv64" fn add_n(number: usize) -> usize {
+>     core::arch::naked_asm!(
+>         "add rdi, {}",
+>         "mov rax, rdi",
+>         "ret",
+>         const 3,
+>     )
+> }
+> # }
+> ```
+
+r[attributes.codegen.naked.syntax]
+The `naked` attribute uses the [MetaWord] syntax and thus does not take any inputs.
+
+r[attributes.codegen.naked.allowed-positions]
+The `naked` attribute may only be applied to:
+
+- [Free functions][items.fn]
+- [Inherent associated functions][items.associated.fn]
+- [Trait impl functions][items.impl.trait]
+- [Trait definition functions][items.traits] with a body
```

Comment:
does this work
```suggestion
- [Trait][items.traits] [associated functions][items.associated.fn] with a body
```

Reply:
This wording can be ambiguous, since it is unclear if it is an associated function in a trait definition or a trait impl. The original wording was specifically chosen to avoid that ambiguity and contrast with the previous item.

Reply:
initially wanted to link this to `items.traits.associated-item-decls`, due to being more specific, because `items.traits` was too general

> - [Trait definition functions][items.traits.associated-item-decls] with a body

is that alright